go/merkletree2/position.go

```diff
@@ -86,8 +93,8 @@ func (t *Tree) getAllSiblings(p *Position) (siblings []Position, parent *Positio
 
 // getDeepestPositionForKey converts the key into the position the key would be
 // stored at if the tree was full with only one key per leaf.
-func (t *Tree) getDeepestPositionForKey(k Key) (*Position, error) {
-	if len(k) != t.cfg.keysByteLength {
+func (t *Config) getDeepestPositionForKey(k Key) (*Position, error) {
```

Comment:
As commented in chat, if you do this: `k = k[0:8]`, you'll get a 2x speedup in generating inclusion proofs.  That relieves almost all the memory pressure caused by bigints, since a bigint with 8 bytes is just a uint64. I think it's worth implementing this optimization for real, it's the lowest hanging fruit to speeding up running this tree.